crates/simulacrum/src/lib.rs

```diff
@@ -95,17 +99,36 @@ where
             .with_chain_start_timestamp_ms(1)
             .deterministic_committee_size(NonZeroUsize::new(1).unwrap())
             .build();
-        let keystore = KeyStore::from_newtork_config(&config);
+        Self::new_with_network_config(&config, rng)
+    }
+
+    pub fn new_with_protocol_version_and_objects(
+        mut rng: R,
+        protocol_version: ProtocolVersion,
+        objects: Vec<Object>,
+    ) -> Self {
+        let config = ConfigBuilder::new_with_temp_dir()
+            .rng(&mut rng)
+            .with_chain_start_timestamp_ms(1)
+            .deterministic_committee_size(NonZeroUsize::new(1).unwrap())
+            .with_protocol_version(protocol_version)
+            .with_objects(objects)
+            .build();
+        Self::new_with_network_config(&config, rng)
+    }
```

Comment:
I don't think we should continue allowing the use of inserting objects into genesis, this is something that has been the bane of many people existence as it breaks a number of safety guarantees like there only being 10B sui.

Reply:
If the purpose is to generate an initial account state with a number of gas objects for different accounts, use the AccountConfig which does the right thing

Reply:
The test service I was going to integrate simulacrum with already current sideloads objects directly into genesis for test validators, as many other tests do.
I've changed the code to use accounts now but if we really want to avoid things such as conservation issues etc, we should fix them in other parts of the code too or risk weirdness.

Reply:
yeah i'm sorry, its been something i've wanted to fix but haven't had an opportunity yet :( Thank you for making this change